README.md

```diff
@@ -1,86 +1,160 @@
+then all the prerequisites are in the folder `depend`, and they can be used in the `cmake` configuration
+as `-DCMAKE_PREFIX_PATH=depend`.
+
+If you have sudo access, as an alternative to the rbuild command, you can install the prerequisites just 
+like in the docker file by calling `./tools/install_prereqs.sh`.
+
+(Note that this script is for Ubuntu. By default, all prerequisites are installed at the default location `/usr/local` 
+and are accessible by all users. For the default location, `sudo` is required to run the script.
+You can also specify a location at which the prerequisites are installed with `./tools/install_prereqs.sh $your_loc`.)
+
+##### Building MIGraphX source and install libs
+
+With the above prerequisites installed, we can build source as:
+
+1) Go to the project folder and create a `build` directory:
 
 
 ```
-CXX=/opt/rocm/llvm/bin/clang++ cmake -DCMAKE_PREFIX_PATH=/some/dir ..
+mkdir build
+cd build
 ```
 
+2) Configure the cmake. If the prerequisites are installed at the default location `/usr/local`, the command is:
 
-#### Changing the cmake configuration
+```
+CXX=/opt/rocm/llvm/bin/clang++ cmake ..
+```
+Otherwise, you need to set `-DCMAKE_PREFIX_PATH=$your_loc` to configure the cmake. 
 
-The configuration can be changed after running cmake by using `ccmake`:
+3) Build MIGraphX source code
 
-` ccmake .. ` **OR** `cmake-gui`: ` cmake-gui ..`
+```
+make -j$(nproc)
+```
 
-## Building the library
+Correctness can be verified as:
 
-The library can be built, from the `build` directory using the 'Release' configuration:
+```
+make -j$(nproc) check
+```
 
-` cmake --build . --config Release ` **OR** ` make `
+MIGraphX libs can be installed as:
 
-And can be installed by using the 'install' target:
+```
+make install
+```
 
-` cmake --build . --config Release --target install ` **OR** ` make install `
+#### Using docker
 
-This will install the library to the `CMAKE_INSTALL_PREFIX` path that was set.
+The easiest way to setup the development environment is to use docker. With the docker file, you can build a docker image as:
 
-To build a debug version of the library, the cmake variable `CMAKE_BUILD_TYPE` can be set to `Debug`.
+    docker build -t migraphx .
 
-` cmake -DCMAKE_BUILD_TYPE=Debug . `
+Then to enter the developement environment use `docker run`:
 
-## Running the tests
+    docker run --device='/dev/kfd' --device='/dev/dri' -v=`pwd`:/code/AMDMIGraphX -w /code/AMDMIGraphX --group-add video -it migraphx
 
-The tests can be run by using the 'check' target:
+In the docker container, all the required prerequisites are already installed, so users can just go to the folder 
+`/code/AMDMIGraphX` and follow the steps in the above [Build MIGraphX source and install
+libs](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/tree/refine_readme#building-migraphx-source-and-install-libs)
```

Comment:
Does this link change when it eventually  gets merged in? If not, we need another way to reference the readme

Reply:
If I remember correctly, it will change accordingly after merging.

Reply:
I think you can link it as `[Build MIGraphX source and install libs](#building-migraphx-source-and-install-libs)` so it links to itself:

https://stackoverflow.com/a/16426829/375343

Reply:
@scxiao are we changing this based off @pfultz2's suggestion?

Reply:
Thanks, fixed.